source/adapters/cuda/program.cpp

```diff
@@ -165,6 +165,47 @@ ur_result_t getKernelNames(ur_program_handle_t) {
   return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
 }
 
+/// Loads images from a list of PTX or CUBIN binaries.
+/// Note: No calls to CUDA driver API in this function, only store binaries
+/// for later.
+///
+/// Note: Only supports one device
```

Comment:
@hdelan is that something we need to think about with multi device context (also the comment at the bottom of the diff)?

Reply:
This should be OK. The entry point has a Device param, so multi dev ctx just changes constructor of program to take the device as well